core/src/main/java/org/axonframework/queryhandling/annotation/MethodQueryMessageHandlerDefinition.java

```diff
@@ -38,10 +44,62 @@
     @SuppressWarnings("unchecked")
     @Override
     public <T> MessageHandlingMember<T> wrapHandler(MessageHandlingMember<T> original) {
-        return original.annotationAttributes(QueryHandler.class)
-                       .map(attr -> (MessageHandlingMember<T>)
-                               new MethodQueryMessageHandlerDefinition.MethodQueryMessageHandlingMember(original, attr))
-                       .orElse(original);
+        Optional<Method> unwrappedMethod = original.unwrap(Method.class);
+        return unwrappedMethod.map(method -> Arrays.stream(method.getParameters())
+                                                   .filter(parameter -> parameter.getType()
+                                                                                 .equals(QueryUpdateEmitter.class))
+                                                   .findFirst()
+                                                   .map(parameter -> original.annotationAttributes(QueryHandler.class)
```

Comment:
Would you mind putting the lambda/function call in this `map()` function on line 52 in a private function? I think that would make it a little more readable.